thoth-api/src/graphql/model.rs

```diff
@@ -1585,6 +1632,28 @@ impl MutationRoot {
         Language::create(&context.db, &data).map_err(|e| e.into())
     }
 
+    #[graphql(description = "Create a new title with the specified values")]
+    fn create_title(
+        context: &Context,
+        #[graphql(description = "Values for title to be created")] data: NewTitle,
+    ) -> FieldResult<Title> {
+        context.token.jwt.as_ref().ok_or(ThothError::Unauthorised)?;
+        context
+            .account_access
+            .can_edit(publisher_id_from_work_id(&context.db, data.work_id)?)?;
+
+        let has_canonical_title = Work::from_id(&context.db, &data.work_id)?
+            .title(context)
+            .is_ok();
+
+        // Only superusers can update the canonical location when a Thoth Location Platform canonical location already exists
+        if has_canonical_title && data.canonical && !context.account_access.is_superuser {
+            return Err(ThothError::ThothUpdateCanonicalError.into());
```

Comment:
This error at present is specific to `location`, so we'll need one for `title` (and rename the existing one)